src/pages/membership/components/youthProfileForm/YouthProfileForm.tsx

```diff
@@ -214,34 +302,147 @@ function YouthProfileForm(componentProps: Props) {
+                          <Field
+                            className={styles.formInputPostal}
+                            as={TextInput}
+                            id={`addresses.${index}.postalCode`}
+                            name={`addresses.${index}.postalCode`}
+                            invalid={isInvalid(
+                              props,
+                              `addresses.${index}.postalCode`
+                            )}
+                            helperText={getErrorCode(
+                              props,
+                              `addresses.${index}.postalCode`
+                            )}
+                            inputMode={
+                              props.values?.primaryAddress?.countryCode === 'FI'
+                                ? 'numeric'
+                                : 'text'
+                            }
+                            labelText={t('registration.postalCode')}
+                          />
+                          <Field
+                            className={styles.formInputCity}
+                            as={TextInput}
+                            id={`addresses.${index}.city`}
+                            name={`addresses.${index}.city`}
+                            invalid={isInvalid(
+                              props,
+                              `addresses.${index}.city`
+                            )}
+                            helperText={getErrorCode(
+                              props,
+                              `addresses.${index}.city`
+                            )}
+                            labelText={t('registration.city')}
+                          />
+                        </div>
+                      </div>
+                      <button
+                        type="button"
+                        className={styles.additionalActionButton}
+                        onClick={() => arrayHelpers.remove(index)}
+                      >
+                        {t('registration.remove')}
+                      </button>
+                    </React.Fragment>
+                  ))}
+                  <br />
+                  <Button
+                    type="button"
+                    iconLeft={<IconPlusCircle />}
+                    className={styles.addAdditional}
+                    variant="supplementary"
+                    onClick={() =>
+                      arrayHelpers.push({
+                        address: '',
+                        postalCode: '',
+                        countryCode: 'FI',
+                        city: '',
+                        primary: false,
+                        addressType: AddressType.OTHER,
+                        __typeName: 'AddressNode',
+                      })
+                    }
+                  >
+                    {t('registration.addAddress')}
+                  </Button>
+                </React.Fragment>
+              )}
+            />
```

Comment:
This file is pretty big now. Wonder if there is a convenient way to split it up?

Reply:
When the layout is updated and information is divided to multiple "boxes",  this gives nice opportunity to split it up.

Reply:
Sounds like a good plan.